### DISCUSSION TITLE
Multiple Collection Fixtures per test class

### DISCUSSION THREAD

**Discussion by magol** [xunit/xunit#2492] [Question] [answered]
**TLDR**
I want the tests in each test class to be run several times with different Collection Fixtures

**Details**
We have many XUnit tests in our project that we use to test our API (each with many `InlineData`). The problem is that we can access a subset of the API via two other connection methods. Today we use a  collection fixture to keep the connection to the main API (it is time consuming to create the connection) and we intend to have three separate collection fixtures, one for each connection path.

I imagine I will create a new attribute I can add to each test class to tell which connection methods it should use. And then it uses the Collection Fixtures that are connected to the various connections. The test runner then run each test in the test classes multiple times, each with different Collection Fixtures. 
Alternatively, maybe I should somehow make it possible to add many `CollectionAttribute` to each test class

I am considering using `[assembly: Xunit.TestFramework ("name", "assembly")]` and implementing my own `XunitTestFramework`. But I can not figure out how to achieve what I want. Do you have any suggestions on how I should proceed?

Ideally, I want the tests for the three collection fixtures to run in parallel while the tests connected to each Collection Fixture run sequentially.

**Comment by bradwilson** [comment, marked as answer]
There is no support for this today. I believe that this is covered by the feature request in #2150.
